.tekton/tssc-cli-pull-request.yaml

```diff
@@ -176,6 +180,65 @@ spec:
       workspaces:
       - name: basic-auth
         workspace: git-auth
+    - name: get-version
+      params:
+      - name: url
+        value: $(params.git-url)
+      - name: revision
+        value: $(params.revision)
+      runAfter:
+      - clone-repository
+      taskSpec:
+        description: Extract version from git repository by cloning it
+        params:
+        - name: url
+          description: Git repository URL
+          type: string
+        - name: revision
+          description: Git revision to checkout
+          type: string
+        results:
+        - name: version
+          description: Git version/tag
+        workspaces:
+        - name: basic-auth
+          description: Workspace for git authentication credentials
+          optional: true
+        steps:
+        - name: clone-and-get-version
+          image: quay.io/konflux-ci/appstudio-utils:ab6b0b8e40e440158e7288c73aff1cf83a2cc8a9@sha256:24179f0efd06c65d16868c2d7eb82573cce8e43533de6cea14fec3b7446e0b14
+          env:
+          - name: GIT_URL
+            value: $(params.url)
+          - name: GIT_REVISION
+            value: $(params.revision)
+          script: |
+            #!/usr/bin/env bash
+            set -eo pipefail
+            
+            # Create temporary directory
+            WORKDIR=$(mktemp -d)
+            cd "$WORKDIR"
+            
+            # Clone repository with tags
+            echo "Cloning repository..."
+            # Use unshallow clone to get full history and tags for git describe
+            git clone --no-single-branch "$GIT_URL" .
+            
+            # Get version using git describe (same logic as Makefile)
+            # Try to get the closest tag, or use commit SHA as fallback
+            VERSION=$(git describe --tags --always 2>/dev/null || echo "v0.0.0-SNAPSHOT")
+            
+            echo "Computed version: $VERSION"
+            printf "%s" "$VERSION" > $(results.version.path)
```

Comment:
Does it clone the repository twice? How it work in comparison to the current approach to clone and build the project?